modules/gojira-core/src/main/java/com/flipkart/gojira/core/aspects/AOPAllianceAdapter.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 Flipkart Internet, pvt ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.gojira.core.aspects;
+
+import org.aopalliance.intercept.ConstructorInterceptor;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.SoftException;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+
+/**
+ * Class that bridges MethodInterceptor of AOPAlliance and AspectJ aspects
+ */
+@Aspect
+public abstract class AOPAllianceAdapter {
+
+    /**
+     * Return the interceptor to use at method execution join points.
+     * Must be overriden by subclasses.
+     *
+     * @return MethodInterceptor, or null if no method advice required
+     */
+    protected abstract MethodInterceptor getMethodInterceptor();
+
+    /**
+     * Return the interceptor to use at constructor execution join points.
+     * May be overriden by subclasses.
+     *
+     * @return ConstructorInterceptor, or null if no constructor advice required
+     */
+    protected ConstructorInterceptor getConstructorInterceptor() {
```

Comment:
This can be removed. Ideally interception/mocking of rpc calls should never happen through constructor. 
We may be providing the support in guice based aop but I did not saw it anywhere getting used.